pkg/tools/gen/genkcl.go

```diff
@@ -0,0 +1,146 @@
+package gen
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+)
+
+// TODO: is need to parse from Comment. Now get kcl schema from Tag
+type GenKclOptions struct {
+	KclPackage       string
+	ParseFromComment bool
+	ParseFromTag     bool
+}
+
+type kclGenerator struct {
+	opts *GenKclOptions
+}
+
+func GenKcl(w io.Writer, filename string, src interface{}, opts *GenKclOptions) error {
```

Comment:
Would U please add more documents and samples in the doc strings for public functions?

Reply:
yep!